### DISCUSSION TITLE
corock님 미션 코드 관련 질문입니다.

### DISCUSSION THREAD

**Discussion by snoopingh** [cowaine/is-my-code-that-strange#96] [Chapter 09-10 (악마, 이름 설계)] [answered]
궁금한 부분 관련해서 댓글을 남겨두었습니다
https://github.com/cowaine/is-my-code-that-strange/commit/8c8752fef62a5c25ee1350c4e0eeb92ff5640403

**Comment by corock** [comment, marked as answer]
> Bank 클래스에서 이자율을 상수로 만들지 않은 이유가 있습니까?
> (private static final float DEFAULT_INTEREST_RATE = 0.03f;)

- #91 이슈에서 '기능 구현이 완료된 AS-IS 버전을 분석하고 리팩터링하기' 라고 소개했습니다.
- #93 코드는 스터디 내용을 반영한 코드가 아닙니다. (리팩터링한 코드는 수정할 부분이 없지 않을까요? ㅎㅎ)
- 그럼에도 답변을 드리자면 상수로 만드는 것도 방법이 될 수 있고, 이자율을 의미하는 `InterestRate` 클래스로 설계해 볼 수도 있을 것 같습니다.

<br>

> 요구사항을 보면 아래와 같은 문구가 있습니다.
> "뱅크에서 엔화는 지원하지 않는 통화입니다. 예를 들어, 엔화 입금 시도 시 다음과 같은 오류 메시지를 출력합니다."
>
> Money 클래스의 생성자에서 통화에 대한 검증을 하고 있네요
> 계좌 생성시에 돈을 입금하면서 발생한 유효하지 않은 통화에 대한 검증은 Account나 Bank에 있는게 더 적절해 보이는데
> @corock 님은 어떻게 생각하십니까?
>
> 제가 미션을 제대로 이해하지 못한 것 일지도 모르겠습니다만
> 입금하려는 돈, 은행 계좌의 잔액 등을 Money 클래스 하나로 처리하신 것을 보고 생긴 의문 같습니다.
> 이것이 10장의 상품 클래스랑 비슷한 맥락일까요? 어렵군요

좋은 태클이네요. 다시 코드를 작성한다면 통화 검증에 대한 책임은 `Bank` 클래스에서 할 것 같습니다.
엔화를 지원하지 않는 건 A 뱅크만의 비즈니스 요건이기 때문에 계좌보다는 은행이 적절하다고 생각합니다.
